ASPNetCoreMastersToDoList/ASPNetCoreMastersToDoList.API/Controllers/UsersController.cs

```diff
@@ -1,22 +1,124 @@
 ﻿using ASPNetCoreMastersToDoList.API.BindingModels;
+using ASPNetCoreMastersToDoList.Data;
+using ASPNetCoreMastersToDoList.Service;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
 
 namespace ASPNetCoreMastersToDoList.API.Controllers
 {
     public class UsersController : ControllerBase
     {
-        private Authentication _authenticationSettings;
 
-        public UsersController(IOptions<Authentication> authenticationSettings)
+        private readonly IUserManagerService _userManagerService;
+        private readonly JwtOptions _options;
+
+        public UsersController(
+            IUserManagerService userManagerService,
+            IOptions<JwtOptions> options)
+        {
+            _userManagerService = userManagerService;
+            _options = options.Value;
+        }
+
+        [HttpPost("Users/Login")]
+        public async Task<IActionResult> Login(LoginBindingModel loginBindingModel)
+        {
+            IActionResult actionResult;
+            var user = await _userManagerService.FindByEmailAsync(loginBindingModel.Email);
+            if (user == null)
+            {
+                actionResult = NotFound(new { error = new[] { $"Email {loginBindingModel.Email} not found" } });
+            }
+            else if (await _userManagerService.CheckPassWordAsync(user, loginBindingModel.Password))
+            {
+                if (!user.EmailConfirmed)
+                {
+                    actionResult = NotFound(new { error = new[] { $"Email {loginBindingModel.Email} not confirmed" } });
+                }
+                else
+                {
+                    var token = GenerateToken(user);
+                    actionResult = Ok(token);
+                }
+            }
+            else
+            {
+                actionResult = NotFound(new { error = new[] { $"Invalid password" } });
+            }
+            return actionResult;
+        }
+
+        [HttpGet("Users/Register")]
```

Comment:
No actions needed, Avoid using HttpGet for "Register" action methods, Thanks!

Reply:
noted. maybe typo on GET but register. MB. :D